doc/design/s3-object-lock-legal-holds.md

```diff
@@ -0,0 +1,102 @@
+# S3 Object Lock Legal Holds
+
+
+## Terminology
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Why make published blobs immutable?
+
+The DANDI Archive's core value proposition is hosting reliable, citable scientific data. Once a dandiset is published and assigned a DOI, the data it references MUST remain immutable to preserve the integrity of scientific citations and reproducibility. Users who cite a published dandiset expect that the data will remain exactly as it was at the time of publication.
+
+Currently, while our application logic prevents modification of published assets, there is no infrastructure-level enforcement.  Adding S3-level immutability provides stronger data safety.
+
+## Requirements
+
+- Published asset blobs MUST be immutable at the S3 infrastructure level
+- The immutability mechanism MUST not interfere with the existing trailing delete functionality (S3 Bucket Versioning)
+- The solution MUST be compatible with future garbage collection features
+- The mechanism SHOULD be efficient when applied to large numbers of blobs during publication
+- There MUST be some sort of "backdoor" that allows deletion of published blobs if necessary, in case of extreme circumstances that warrant deletion of a published asset and/or dandiset
+
+## Proposed Solution
+
+AWS S3 Object Lock with legal holds provides exactly the protection we need. A legal hold is a flag that can be applied to an S3 object version to prevent deletion or modification, regardless of any retention policies or IAM permissions. Unlike retention modes, legal holds have no expiration date and remain in effect until explicitly removed via the `s3:PutObjectLegalHold` S3 API. See the [documentation for S3 legal holds](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html#object-lock-legal-holds) for more details.
+
+### S3 Object Lock
+
+S3 Object Lock is a feature that prevents object versions from being deleted or overwritten for a specified period or indefinitely. It operates in three ways:
+
+1. **Governance mode**: Prevents most users from deleting object versions, but users with special permissions can override
+2. **Compliance mode**: No user can delete the object version until the retention period expires, including the root account
+3. **Legal hold**: A separate flag independent of retention mode that prevents deletion until removed
+
+Goverence and compliance modes operate at the bucket level, while legal holds operate at the object level.
```

Comment:
```suggestion
Goverence and compliance modes operate at the bucket level, while legal holds operate at the object version level.
```

IMHO an important feature (if I got it right) since will allow later to protect Zarr versions. (added below as an item)